### ISSUE TITLE
Add content to the "let" section of the language guide that talks about destructuring

### ISSUE THREAD

**Issue by CrowdHailer** [gleam-lang/suggestions#130]
https://gleam.run/book/tour/let-bindings.html

I'm not sure if this should just be a link to the case section or if it's different enough to have it's own content. Given the hypothetical future of exhaustiveness checking it will be a bit different because only single record varients can be destructured in a let.

**Comment by lpil** [comment]
I think this should go in the sections for each data type as if in let it would mean using data structures before they are introduced.

**Comment by CrowdHailer** [comment]
Fair enough. I reckon a link is still worth having because I don't think it's mentioned that you can destructure in let's anywhere.
Unless it's a deliberate choice to lead people to destructuring with assert.
